sherpa/astro/data.py

```diff
@@ -107,7 +107,7 @@ def __str__(self):
         old = self._fields
```

Comment:
You just reminded me the comment I wanted to add to the description of the fix. I had your same impression and urge. The code does look the same, and even without adding an intermediate class the __str__ method might be pushed up. I'll have to check that the code is exactly the same, though.